### PR TITLE
Remove configuration form validation groups

### PR DESCRIPTION
**Pull request by pjedrzejewski**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | https://github.com/Sylius/Sylius/issues/1295
| License       | MIT
| Doc PR        | -

Based on #3721.

Helloes.

This change basically forces all Sylius resource forms to use Default validation group. Result:

* On (promotion action/rule, calculator, report etc.) configuration forms we can safely use constraints without any groups, so can safely get rid of these ugly and long configuration nodes for these forms. Use-case where someone wants to override validation of FixedDiscountConfigurationType is super rare and we still allow this by overriding whole form. Supporting this rare use-case with current approach is not worth all the complexity that is introduced.
* DX is improved - developer does not need to bother/remember/read in documentation that he needs to add "sylius" validation group to overridden entity in their custom app. He can just add constraint and be done with it.
* All Sylius constraints use proper group "sylius" and ``validation_groups`` config node is available for every single resource, thus we maintain the ability to completely rework validation.

I did that for Promotion only right now, but will do this for everything else if I get support from you guys. :) Let me know what do you think!

Will be easier to review once #3721 is merged. Perhaps wait with reviewing the code.

### CODE REVIEW

**Comment by aRn0D** [comment]
@pjedrzejewski I think you're right! This feature is an overkill, I guest it is use rarely...

**Comment by michalmarcinkowski** [comment]
I like cleanups! Thanks Paweł! :+1:

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php [hunk 1]
```diff
@@ -154,16 +154,15 @@ protected function addForms(ContainerBuilder $container, MetadataInterface $meta
...
-                        $validationGroups = new Parameter($validationGroupsParameterName);
-                    } else {
+                    if (!$container->hasParameter($validationGroupsParameterName)) {
                         $validationGroups = array('Default');
                     }
 
                     $definition->setArguments(array(
                         $metadata->getClass('model'),
-                        $validationGroups,
+                        $validationGroups
```
Comment (stloyd) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php]: IMO this should be reverted ;)

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php [hunk 2]
```diff
@@ -154,16 +154,15 @@ protected function addForms(ContainerBuilder $container, MetadataInterface $meta
 
                 default:
                     $validationGroupsParameterName = sprintf('%s.validation_groups.%s%s', $metadata->getApplicationName(), $metadata->getName(), $suffix);
+                    $validationGroups = new Parameter($validationGroupsParameterName);
```
Comment (stloyd) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php]: This should be on `} else {` IMO.

Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php]: I somehow missed that file... IMO this should be changed to:
```php
$validationGroups = array('Default');

if ($container->hasParameter($validationGroupsParameterName) {
   $validationGroups = new Parameter($validationGroupsParameterName);
}
```

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php (hunk 1 above)
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php]: :+1:

**Pull request merged**